modules/snooker/src/main/ts/ephox/snooker/model/RunOperation.ts

```diff
@@ -169,6 +169,15 @@ const onCells = (warehouse: Warehouse, target: TargetSelection): Optional<Struct
 
 // Custom unlocked extractors
 
+const onUnlockedRows = (warehouse: Warehouse, target: TargetSelection): Optional<Structs.DetailExt[]> =>
```

Comment:
This is a nitpick but in Snooker, putting `contenteditable="false"` doesn't mean the cell/column/row is locked. Snooker only has the concept of locked columns (`data-snooker-locked-cols`) but not locked rows at this stage.

If we make it so that a row cannot have its type changed if the `tr` has `contenteditable="false"` then we would also have to consider what should happen when a `col` has `contenteditable="false"` for various operations as well.

For the row type table operations, I would be more inclined to just use `RunOperation.onCell`  and `RunOperation.onCells` and re-evaluate when locked rows are properly implemented some day.

Reply:
I had some concerns that this might then allow changes on content which is CEF entirely, rather than where the cell is CEF but the table/row as a whole is CET.
As for when we need to consider td's being CEF, isn't that what the whole isLocked check is for?

Reply:
`isLocked` is tied to the locked columns. For example, if we have `data-snooker-locked-cols="0"` then all cells in the first column are considered locked but the cells are not necessarily CEF

Reply:
Alright, I'll change it to onCell.